File: conformance/bundles/08-string-capture/GreetSteps.java
package dev.varar.conformance.bundle08;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/** Java sibling of {@code greet.steps.ts} / {@code greet.steps.py} (bundle {@code 08-string-capture}). */
public final class GreetSteps implements StepDefinitions<GreetSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.defineState(Ctx::new);

        s.stimulus("I greet {string}", (Ctx ctx, String name) -> ctx);
    }
}

File: conformance/bundles/08-string-capture/greet.steps.kt
@file:JvmName("GreetSteps")

// Kotlin sibling of greet.steps.ts / greet.steps.py / GreetSteps.java (bundle
// 08-string-capture).
package dev.varar.kotlin.conformance.bundle08

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I greet {string}") { name: String -> this }
}
